include/CXXGraph/Graph/Algorithm/Dinic_impl.hpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <queue>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> capacities;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> flow;
+    
+    bool bfs(const CXXGraph::id_t& source, const CXXGraph::id_t& sink, std::unordered_map<CXXGraph::id_t, int>& distance) {
```

Comment:
Thanks for implementing `bfs` and `dfs` (and Dinic's!).

These actually seem generally useful. Should we consider making them public? (assuming the storage requirement would need to change given the above comment).